src/components/Stream/TopologyList.tsx

```diff
@@ -20,13 +24,24 @@ interface ParamTypes {
   nodeId: string
 }
 
+const PAGE_SIZE = 10
+
 const TopologyList = ({ id }: Props) => {
   const { visibleNodes, stream } = useStore()
   const { nodeId: encodedNodeId } = useParams<ParamTypes>()
   const history = useHistory()
   const listRef = useRef<HTMLDivElement>(null)
   const scrollTimeout = useRef<number | undefined>(undefined)
   const isMounted = useIsMounted()
+  const [pageChangedOnLoad, setPageChangedOnLoad] = useState(false)
+  const { loadNodeLocations } = useController()
```

Comment:
There is a small hiccup always when I change to next page. Should we change `loadNodeLocations` implementation to call `updateLocations` after each result is received. Then we didn't need to wait for all results to be ready.

Reply:
Hmm, one by one? Yeah, that might work 👍